cmd/nitro/init.go

```diff
@@ -937,6 +945,22 @@ func openInitializeExecutionDB(ctx context.Context, stack *node.Node, config *No
 	return rebuildLocalWasm(ctx, &config.Execution, l2BlockChain, executionDB, wasmDB, config.Init.RebuildLocalWasm)
 }
 
+// resolveInitialL1BaseFee reads the initial L1 base fee according to the following rules:
+// 1. If there is no genesisArbOSInit, return the value from chainConfig (in case it is not set either, it should fall back to default).
+// 2. If there is a genesisArbOSInit, and chainConfig.InitialL1BaseFee is set, ensure they match.
+// 3. If there is a genesisArbOSInit, and chainConfig.InitialL1BaseFee is not set, return the value from genesisArbOSInit.
+// Assumes that chainConfig is not nil.
+func resolveInitialL1BaseFee(genesisArbOSInit *params.ArbOSInit, chainConfig *conf.L2Config) (*big.Int, error) {
```

Comment:
How about adding a test related to this func?

Reply:
I added test directly for this function in [800bf40](https://github.com/OffchainLabs/nitro/pull/4171/commits/800bf4009da24c47ec5e5754c75fc746b560e0d6); otherwise, testing whole `openInitializeExecutionDB` would be really complex - we would have to inspect written genesis block data and provide a lot of other configuration options